test/src/test/java/org/corfudb/infrastructure/LogUnitServerTest.java

```diff
@@ -209,5 +208,90 @@ public void testVerifyWithNoVerifyLog() throws Exception {
         ServerContext context = builder.build();
         LogUnitServer logunit = new LogUnitServer(context);
     }
+
+
+    @Test
+    public void checkOverwriteExceptionIsNotThrownWhenTheRankIsHigher() throws Exception {
+        String serviceDir = PARAMETERS.TEST_TEMP_DIR;
+
+        LogUnitServer s1 = new LogUnitServer(new ServerContextBuilder()
+                .setLogPath(serviceDir)
+                .setMemory(false)
+                .build());
+
+        this.router.reset();
+        this.router.addServer(s1);
+
+        final long ADDRESS_0 = 0L;
+        final long ADDRESS_1 = 100L;
+        //write at 0
+        ByteBuf b = ByteBufAllocator.DEFAULT.buffer();
+        Serializers.CORFU.serialize("0".getBytes(), b);
+        WriteRequest m = WriteRequest.builder()
+                .writeMode(WriteMode.NORMAL)
+                .data(new LogData(DataType.DATA, b))
+                .build();
+        m.setGlobalAddress(ADDRESS_0);
+        m.setStreams(Collections.EMPTY_SET);
+        m.setRank(new IMetadata.DataRank(0));
+        m.setBackpointerMap(Collections.emptyMap());
+        sendMessage(CorfuMsgType.WRITE.payloadMsg(m));
+
+        assertThat(s1)
+                .containsDataAtAddress(ADDRESS_0);
+        assertThat(s1)
+                .isEmptyAtAddress(ADDRESS_1);
+
+
+        // repeat: do not throw exception, the overwrite is forced
+        b.clear();
+        b = ByteBufAllocator.DEFAULT.buffer();
+        Serializers.CORFU.serialize("1".getBytes(), b);
+        m = WriteRequest.builder()
+                .writeMode(WriteMode.NORMAL)
+                .data(new LogData(DataType.DATA, b))
+                .build();
+        m.setGlobalAddress(ADDRESS_0);
+        m.setStreams(Collections.EMPTY_SET);
+        m.setBackpointerMap(Collections.emptyMap());
+
+
+        WriteRequest m2 = WriteRequest.builder()
+                .writeMode(WriteMode.NORMAL)
+                .data(new LogData(DataType.DATA, b))
```

Comment:
Just reminder - we decided, that this code is correct, but I renamed the data type from PROPOSAL to RANK_ONLY for better readability.